.github/workflows/ci-pipeline.yml

```diff
@@ -138,25 +138,32 @@ jobs:
       contains(github.event.pull_request.labels.*.name, 'auto-merge') &&
       github.event.pull_request.user.login == 'bitnami-bot'
     steps:
-      - name: Enable auto-merge feature
-        if: ${{ needs.vib-verify.result == 'success' }}
-        uses: reitermarkus/automerge@v2.2.0
-        with:
-          # Necessary to trigger CD workflow
-          token: ${{ secrets.BITNAMI_BOT_TOKEN }}
-          merge-method: squash
-          pull-request: ${{ github.event.number }}
       # Approve the CI's PR if the 'VIB Verify' job succeeded
       # Approved by the 'github-actions' user; a PR can't be approved by its author
       - name: PR approval
         if: ${{ needs.vib-verify.result == 'success' }}
         uses: hmarr/auto-approve-action@v3.0.0
         with:
           pull-request-number: ${{ github.event.number }}
+      - name: Merge
+        id: merge
+        if: ${{ needs.vib-verify.result == 'success' }}
+        uses: actions/github-script@v6
+        with:
+          result-encoding: string
+          retries: 3
+          github-token: ${{ secrets.BITNAMI_BOT_TOKEN }}
+          script: |
+            github.rest.pulls.merge({
+              pull_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              merge_method: 'squash'
+            })
       # If the CI did not succeed ('VIB Verify' failed or skipped),
       # post a comment on the PR and assign a maintainer agent to review it
       - name: Manual review required
-        if: ${{ needs.vib-verify.result != 'success' }}
+        if: ${{ always() && (needs.vib-verify.result != 'success' || steps.merge.outcome != 'success' ) }}
```

Comment:
Do you need to use the `always()` statement here? AFAIK, it was used at **job** level when it defined some dependencies (with `needs`) but you wanted to run it regardless of their status. Maybe I am wrong, but why is it needed here?

Reply:
If the merge step fails, the `Manual review required` step will be skipped. To avoid that behaviour, we have to use the always() statement. This [execution](https://github.com/fmulero/charts/actions/runs/3678152664/jobs/6221915489) was triggered before this change and it failed, there you can see how the `Manual review required` step was skipped.